From c8e58304bf4b1b327067bd92ca6e3c9db897c572 Mon Sep 17 00:00:00 2001
From: Ryan Roden-Corrent <ryan@rcorre.net>
Date: Sun, 7 Jan 2024 21:08:41 -0500
Subject: [PATCH 170/188] Add textobject queries for protobuf grammar. (#9184)

Given `message Foo {string s = 1;}`
- `mat` selects `message Foo {string s = 1}`
- `mit` selects `{string s = 1;}`

Given `service SearchService { rpc Search(Req) returns (Resp); }
- `mit` or `mat` selects `Req` or `Resp`
- `mif` or `maf` selects `rpc Search(Req) returns (Resp);`
- `mit` selects { rpc Search(Req) returns (Resp); }`
- `mat` selects `service SearchService { rpc Search(Req) returns (Resp); }`
---
 book/src/generated/lang-support.md       | 2 +-
 runtime/queries/protobuf/textobjects.scm | 9 +++++++++
 2 files changed, 10 insertions(+), 1 deletion(-)
 create mode 100644 runtime/queries/protobuf/textobjects.scm

--- a/book/src/generated/lang-support.md
+++ b/book/src/generated/lang-support.md
@@ -128,7 +128,7 @@
 | ponylang | ✓ | ✓ | ✓ |  |
 | prisma | ✓ |  |  | `prisma-language-server` |
 | prolog |  |  |  | `swipl` |
-| protobuf | ✓ |  | ✓ | `bufls`, `pb` |
+| protobuf | ✓ | ✓ | ✓ | `bufls`, `pb` |
 | prql | ✓ |  |  |  |
 | purescript | ✓ | ✓ |  | `purescript-language-server` |
 | python | ✓ | ✓ | ✓ | `pylsp` |
--- /dev/null
+++ b/runtime/queries/protobuf/textobjects.scm
@@ -0,0 +1,9 @@
+(message (messageBody) @class.inside) @class.around
+(enum (enumBody) @class.inside) @class.around
+(service (serviceBody) @class.inside) @class.around
+
+(rpc (enumMessageType) @parameter.inside) @function.inside
+(rpc (enumMessageType) @parameter.around) @function.around
+
+(comment) @comment.inside
+(comment)+ @comment.around
